### ISSUE TITLE
Debug - unable to Step Into or see variable values

### ISSUE THREAD

**Issue by abirtley** [wallabyjs/public#3179]
### Issue description or question

Wallaby tests are running, and light up the lines red or green as expected.

I can also Show Line Values and Show Line Timings without any problems:

<img width="647" alt="Screen Shot 2023-03-02 at 11 10 30 am" src="https://user-images.githubusercontent.com/19584280/222296907-d8693900-fd49-47d2-9b6c-2b6f26874f2a.png">

However, when debugging, I cannot Step Into, nor can I see values (either by hovering or by looking at the value explorer). You can see both problems in the videos below. (Notice that Wallaby Debug does step into the anonymous compare function, but not into the `selectBest` function which is the one under test.)

https://user-images.githubusercontent.com/19584280/222297052-544e511d-85e6-4a31-8bf6-f568a657c401.mov

https://user-images.githubusercontent.com/19584280/222297070-541f7211-1251-4c87-9c60-5cd709f5c991.mov

Please let me know what other information I can provide to assist with tracking this down.

### Wallaby diagnostics report

```
{
  editorVersion: '1.75.1',
  pluginVersion: '1.0.349',
  editorType: 'VSCode',
  osVersion: 'darwin 21.6.0',
  nodeVersion: 'v16.14.0',
  coreVersion: '1.0.1386',
  checksum: 'MDI0ODgzYWU0ZGU0OWE4MzlkZTdmZjNiMWZkNjMwZDIsMTY3ODMyMDAwMDAwMCww',
  config: {
    testFramework: { version: 'jest@24.8.0', configurator: 'jest@24.8.0', reporter: 'jest@24.8.0', starter: 'jest@24.8.0', configFile: './jest.config.js', autoDetected: true },
    diagnostics: {
      jest: {
        config: {
          configs: [
            {
              automock: false,
              cache: true,
              cacheDirectory: '/private/var/folders/kn/lrcbwj4x5vzfrhw3ln56yjnh0000gn/T/jest_dx',
              clearMocks: false,
              coveragePathIgnorePatterns: [ '/node_modules/' ],
              cwd: '<homeDir>/Sites/projectName/backend',
              dependencyExtractor: undefined,
              detectLeaks: false,
              detectOpenHandles: false,
              displayName: undefined,
              errorOnDeprecated: false,
              extensionsToTreatAsEsm: [],
              extraGlobals: [],
              filter: undefined,
              forceCoverageMatch: [],
              globalSetup: undefined,
              globalTeardown: undefined,
              globals: {},
              haste: { computeSha1: false, enableSymlinks: false, forceNodeFilesystemAPI: false, throwOnModuleCollision: false },
              injectGlobals: true,
              moduleDirectories: [ 'node_modules' ],
              moduleFileExtensions: [ 'js', 'jsx', 'ts', 'tsx', 'json', 'node' ],
              moduleLoader: undefined,
              moduleNameMapper: [
                [ '@projectName/common/(.*)', '<homeDir>/Sites/projectName/common/$1' ],
                [ 'true-myth/result', '<homeDir>/Sites/projectName/node_modules/true-myth/dist/result.js' ],
                [ 'true-myth/maybe', '<homeDir>/Sites/projectName/node_modules/true-myth/dist/maybe.js' ]
              ],
              modulePathIgnorePatterns: [],
              modulePaths: undefined,
              name: 'b7a79501ed5975f2c25b1c79857190fc',
              prettierPath: 'prettier',
              resetMocks: false,
              resetModules: false,
              resolver: undefined,
              restoreMocks: false,
              rootDir: '<homeDir>/Sites/projectName/backend',
              roots: [ '<homeDir>/Sites/projectName/backend/test' ],
              runner: '<homeDir>/Sites/projectName/node_modules/jest-runner/build/index.js',
              setupFiles: [],
              setupFilesAfterEnv: [ '<homeDir>/Sites/projectName/backend/test/jest.setup.ts' ],
              skipFilter: false,
              skipNodeResolution: undefined,
              slowTestThreshold: 5,
              snapshotFormat: undefined,
              snapshotResolver: undefined,
              snapshotSerializers: [],
              testEnvironment: '<homeDir>/Sites/projectName/node_modules/jest-environment-node/build/index.js',
              testEnvironmentOptions: {},
              testLocationInResults: false,
              testMatch: [ '**/*.*test.(ts|tsx|js)' ],
              testPathIgnorePatterns: [ '/node_modules/' ],
              testRegex: [],
              testRunner: '<homeDir>/Sites/projectName/node_modules/jest-circus/runner.js',
              testURL: 'http://localhost',
              timers: 'real',
              transform: [
                [ '\\.ts$', '<homeDir>/Sites/projectName/node_modules/esbuild-jest/esbuild-jest.js', { sourcemap: true } ],
                [ 'true-myth[\\/].*\\.js$', '<homeDir>/Sites/projectName/node_modules/esbuild-jest/esbuild-jest.js', { sourcemap: true } ]
              ],
              transformIgnorePatterns: [ '/node_modules/(?!(true-myth)/)' ],
              unmockedModulePathPatterns: undefined,
              watchPathIgnorePatterns: []
            }
          ],
          globalConfig: {
            bail: 0,
            changedFilesWithAncestor: false,
            changedSince: undefined,
            collectCoverage: true,
            collectCoverageFrom: [ '(functions|core|services)/**/*.{ts,js}' ],
            collectCoverageOnlyFrom: undefined,
            coverageDirectory: '<homeDir>/Sites/projectName/backend/coverage',
            coverageProvider: 'babel',
            coverageReporters: [ 'json', 'text', 'lcov', 'clover' ],
            coverageThreshold: undefined,
            detectLeaks: false,
            detectOpenHandles: false,
            errorOnDeprecated: false,
            expand: false,
            filter: undefined,
            findRelatedTests: false,
            forceExit: false,
            globalSetup: undefined,
            globalTeardown: undefined,
            json: false,
            lastCommit: false,
            listTests: false,
            logHeapUsage: false,
            maxConcurrency: 5,
            maxWorkers: 1,
            noSCM: undefined,
            noStackTrace: false,
            nonFlagArgs: undefined,
            notify: false,
            notifyMode: 'failure-change',
            onlyChanged: false,
            onlyFailures: false,
            outputFile: undefined,
            passWithNoTests: false,
            projects: [],
            replname: undefined,
            reporters: undefined,
            rootDir: '<homeDir>/Sites/projectName/backend',
            runTestsByPath: false,
            silent: undefined,
            skipFilter: false,
            snapshotFormat: undefined,
            testFailureExitCode: 1,
            testNamePattern: undefined,
            testPathPattern: '',
            testResultsProcessor: undefined,
            testSequencer: '<homeDir>/Sites/projectName/node_modules/@jest/test-sequencer/build/index.js',
            testTimeout: undefined,
            updateSnapshot: 'new',
            useStderr: false,
            verbose: undefined,
            watch: false,
            watchAll: false,
            watchPlugins: undefined,
            watchman: true
          },
          hasDeprecationWarnings: false,
          wallaby: {
            roots: [ 'test' ],
            watchPathIgnorePatterns: [ '/node_modules/', '\\./dist/|\\./build/|\\./coverage/|\\./git/|/\\..+/', '/private/var/folders/kn/lrcbwj4x5vzfrhw3ln56yjnh0000gn/T/jest_dx', '\\./coverage' ],
            testPathIgnorePatterns: [ '/node_modules/', '\\./dist/|\\./build/|\\./coverage/|\\./git/|/\\..+/', '/private/var/folders/kn/lrcbwj4x5vzfrhw3ln56yjnh0000gn/T/jest_dx', '\\./coverage' ],
            testMatch: [ '**/*.*test.(ts|tsx|js)' ],
            testRegex: []
          }
        }
      }
    },
    filesWithCoverageCalculated: [ '(functions|core|services)/**/*.{ts,js}' ],
    filesWithNoCoverageCalculated: [],
    globalSetup: false,
    dot: true,
    files: [
      { pattern: '/node_modules/', regexp: /\/node_modules\//, ignore: true, trigger: true, load: true, file: true, test: true },
      {
        pattern: '\\./dist/|\\./build/|\\./coverage/|\\./git/|/\\..+/',
        regexp: /\.\/dist\/|\.\/build\/|\.\/coverage\/|\.\/git\/|\/\..+\//,
        ignore: true,
        trigger: true,
        load: true,
        file: true,
        test: true
      },
      {
        pattern: '/private/var/folders/kn/lrcbwj4x5vzfrhw3ln56yjnh0000gn/T/jest_dx',
        regexp: /\/private\/var\/folders\/kn\/lrcbwj4x5vzfrhw3ln56yjnh0000gn\/T\/jest_dx/,
        ignore: true,
        trigger: true,
        load: true,
        file: true,
        test: true
      },
      { pattern: '\\./coverage', regexp: /\.\/coverage/, ignore: true, trigger: true, load: true, file: true, test: true },
      { pattern: '**/**', ignore: false, trigger: true, load: true, order: 1 },
      { pattern: 'test/**', ignore: false, trigger: true, load: true, order: 2 },
      { pattern: 'test/**/*.*test.(ts|tsx|js)', ignore: true, trigger: true, load: true, file: true }
    ],
    tests: [
      { pattern: '/node_modules/', regexp: /\/node_modules\//, ignore: true, trigger: true, load: true, test: true, file: false },
      {
        pattern: '\\./dist/|\\./build/|\\./coverage/|\\./git/|/\\..+/',
        regexp: /\.\/dist\/|\.\/build\/|\.\/coverage\/|\.\/git\/|\/\..+\//,
        ignore: true,
        trigger: true,
        load: true,
        test: true,
        file: false
      },
      {
        pattern: '/private/var/folders/kn/lrcbwj4x5vzfrhw3ln56yjnh0000gn/T/jest_dx',
        regexp: /\/private\/var\/folders\/kn\/lrcbwj4x5vzfrhw3ln56yjnh0000gn\/T\/jest_dx/,
        ignore: true,
        trigger: true,
        load: true,
        test: true,
        file: false
      },
      { pattern: '\\./coverage', regexp: /\.\/coverage/, ignore: true, trigger: true, load: true, test: true, file: false },
      { pattern: 'test/**/*.*test.(ts|tsx|js)', ignore: false, trigger: true, load: true, test: true, order: 3 }
    ],
    runAllTestsInAffectedTestFile: false,
    updateNoMoreThanOneSnapshotPerTestFileRun: false,
    compilers: {},
    smartStart: [ { test: true, source: false, pattern: '**/*', startMode: 'open' }, { test: false, source: true, pattern: '**/*', startMode: 'edit' } ],
    logLimits: { inline: { depth: 5, elements: 5000 }, values: { default: { stringLength: 200 }, autoExpand: { elements: 5000, stringLength: 8192, depth: 10 } } },
    preprocessors: {},
    maxConsoleMessagesPerTest: 100,
    autoConsoleLog: true,
    delays: { run: 0, edit: 100, update: 0 },
    workers: { initial: 0, regular: 0, recycle: false },
    teardown: undefined,
    hints: {
      ignoreCoverage: '__REGEXP /ignore coverage|istanbul ignore/',
      ignoreCoverageForFile: '__REGEXP /ignore file coverage/',
      commentAutoLog: '?',
      testFileSelection: { include: '__REGEXP /file\\.only/', exclude: '__REGEXP /file\\.skip/' }
    },
    automaticTestFileSelection: true,
    runSelectedTestsOnly: true,
    mapConsoleMessagesStackTrace: false,
    extensions: {},
    env: {
      type: 'node',
      params: {},
      runner: '<homeDir>/.nvm/versions/node/v16.14.0/bin/node',
      viewportSize: { width: 800, height: 600 },
      options: { width: 800, height: 600 },
      bundle: true
    },
    reportUnhandledPromises: true,
    slowTestThreshold: 75,
    lowCoverageThreshold: 80,
    runAllTestsWhenNoAffectedTests: false,
    configCode: "module.exports = function () {\n  return {\n    autoDetect: true,\n    testFramework: {\n      configFile: './jest.config.js',\n    },\n  };\n};\n",
    testLocations: [ 'test/common/tfh.test.ts' ]
  },
  packageJSON: {
    dependencies: {
      '@middy/core': '^2.5.7',
      '@middy/http-error-handler': '^2.5.7',
      '@middy/validator': '^2.5.7',
      '@sendinblue/client': '^3.1.0',
      '@sentry/integrations': '^7.7.0',
      '@sentry/node': '^7.7.0',
      '@sentry/tracing': '^7.7.0',
      '@types/aws-lambda': '^8.10.83',
      arangojs: '^7.7.0',
      diff: '^5.1.0',
      'english-a-an': '^2.2.2',
      'english-a-an-list': '^2.2.2',
      'middy-middleware-jwt-auth': '^5.0.0',
      nanoid: '^3.3.3',
      'node-fetch': '^2',
      pluralize: '^8.0.0',
      stripe: '^9.1.0',
      'true-myth': '^5.2.0',
      winston: '^3.6.0',
      zod: '^3.19.1'
    },
    devDependencies: {
      '@types/diff': '^5.0.2',
      '@types/node-fetch': '^2',
      '@types/pluralize': '^0.0.29',
      'aws-sdk': '^2.1084.0',
      'aws-testing-library': '^1.1.6',
      esbuild: '^0.16.10',
      'esbuild-jest': '^0.5.0',
      'esbuild-runner': '^2.2.1',
      jest: '^27.2.0',
      'sls-test-tools': '^1.0.7',
      tslib: '^2.3.1'
    }
  },
  fs: { numberOfFiles: 127 },
  debug: [
    '2023-03-02T00:21:37.215Z config Attempting automatic configuration for angular\n',
    '2023-03-02T00:21:37.218Z angular/cli config Angular CLI not found.\n',
    '2023-03-02T00:21:37.218Z config Finished attempting automatic configuration for angular (3ms)\n',
    '2023-03-02T00:21:37.221Z config Attempting automatic configuration for jest\n',
    '2023-03-02T00:21:37.439Z jest/config Detected Jest.\n',
    '2023-03-02T00:21:37.439Z jest/config Configured Jest.\n',
    '2023-03-02T00:21:37.439Z config Finished attempting automatic configuration for jest (218ms)\n',
    '2023-03-02T00:21:37.441Z project Wallaby Node version: v16.14.0\n',
    '2023-03-02T00:21:37.441Z project Wallaby config: <homeDir>/Sites/projectName/backend/wallaby.js\n',
    '2023-03-02T00:21:37.448Z fs File system starting\n',
    '2023-03-02T00:21:37.488Z fs File system scan completed\n',
    '2023-03-02T00:21:37.499Z project File cache: <homeDir>/.vscode/extensions/wallabyjs.wallaby-vscode-1.0.349/projects/eb84c04c9b5e73d5\n',
    '2023-03-02T00:21:37.589Z uiService Listening port 51235\n',
    '2023-03-02T00:21:37.594Z project package.json file change detected, invalidating local cache\n',
    '2023-03-02T00:21:37.612Z workers Parallelism for initial run: 6, for regular run: 3\n',
    '2023-03-02T00:21:37.612Z workers Starting run worker instance #0\n',
    '2023-03-02T00:21:37.612Z workers Starting run worker instance #1\n',
    '2023-03-02T00:21:37.612Z workers Starting run worker instance #2\n',
    '2023-03-02T00:21:37.612Z workers Starting run worker instance #3\n',
    '2023-03-02T00:21:37.612Z workers Starting run worker instance #4\n',
    '2023-03-02T00:21:37.612Z workers Starting run worker instance #5\n',
    '2023-03-02T00:21:37.612Z workers Web server is listening at 65452\n',
    '2023-03-02T00:21:37.613Z project File cache requires some updates, waiting required files from IDE\n',
    '2023-03-02T00:21:37.686Z project Stopping process pool\n',
    '2023-03-02T00:21:37.688Z project Test run started; run priority: 3\n',
    '2023-03-02T00:21:37.689Z project Running all tests\n',
    '2023-03-02T00:21:37.690Z project Test run finished\n',
    '2023-03-02T00:21:37.691Z project Test run data re-queued\n',
    '2023-03-02T00:21:37.801Z project Requested to run all tests\n',
    '2023-03-02T00:21:37.802Z project Test run started; run priority: 3\n',
    '2023-03-02T00:21:37.802Z project Running all tests\n',
    '2023-03-02T00:21:37.804Z workers Starting test run, priority: 3\n',
    '2023-03-02T00:21:37.804Z nodeRunner Starting sandbox [worker #0, session #gnt61]\n',
    '2023-03-02T00:21:37.804Z nodeRunner Preparing sandbox [worker #0, session #gnt61]\n',
    '2023-03-02T00:21:37.870Z workers Started run worker instance (delayed) #1\n',
    '2023-03-02T00:21:37.870Z workers Started run worker instance (delayed) #5\n',
    '2023-03-02T00:21:37.872Z workers Started run worker instance (delayed) #2\n',
    '2023-03-02T00:21:37.873Z workers Started run worker instance (delayed) #4\n',
    '2023-03-02T00:21:37.873Z workers Started run worker instance (delayed) #3\n',
    '2023-03-02T00:21:37.884Z workers Started run worker instance (delayed) #0\n',
    '2023-03-02T00:21:37.884Z nodeRunner Prepared sandbox [worker #0, session #gnt61]\n',
    '2023-03-02T00:21:37.884Z workers [worker #0, session #gnt61] Running tests in sandbox\n',
    '2023-03-02T00:21:39.249Z workers Scheduling Jest Test Run (gnt61): 2023-03-02T00:21:38.181Z\n',
    '2023-03-02T00:21:39.259Z workers [gnt61] Loaded unknown number of test(s)\n',
    '2023-03-02T00:21:39.259Z workers [gnt61] Test executed: works with simple data\n',
    '2023-03-02T00:21:39.259Z workers [gnt61] Test executed: works with complex data\n',
    '2023-03-02T00:21:39.259Z workers [gnt61] Test executed: applies patches in order\n',
    '2023-03-02T00:21:39.260Z workers [gnt61] Test executed: works with sub-objects\n',
    '2023-03-02T00:21:39.260Z workers [gnt61] Test executed: executes the function if input is not undefined\n',
    '2023-03-02T00:21:39.260Z workers [gnt61] Test executed: does not execute the function if input undefined\n',
    '2023-03-02T00:21:39.260Z workers [gnt61] Test executed: returns undefined if the input is undefined\n',
    '2023-03-02T00:21:39.260Z workers [gnt61] Test executed: works with a different input and return type\n',
    '2023-03-02T00:21:39.261Z workers [gnt61] Test executed: works\n',
    '2023-03-02T00:21:39.262Z workers [gnt61] Run 9 test(s), skipped 0 test(s)\n',
    '2023-03-02T00:21:39.262Z workers Jest Test Run Complete (gnt61): 2023-03-02T00:21:39.247Z\n',
    '2023-03-02T00:21:39.263Z workers [gnt61] Sandbox is responsive, closing it\n',
    '2023-03-02T00:21:39.263Z project Test run finished\n',
    '2023-03-02T00:21:39.264Z project Processed console.log entries\n',
    '2023-03-02T00:21:39.264Z project Processed loading sequences\n',
    '2023-03-02T00:21:39.264Z project Processed executed tests\n',
    '2023-03-02T00:21:39.265Z project Processed code coverage\n',
    '2023-03-02T00:21:39.277Z project Test run result processed and sent to IDE\n'
  ]
}
```

**Comment by abirtley** [comment]
One thing that seems strange, straight out of the box, is the following line in the diagnostic dump:

```
testFramework: { version: 'jest@24.8.0', configurator: 'jest@24.8.0', reporter: 'jest@24.8.0', starter: 'jest@24.8.0', configFile: './jest.config.js', autoDetected: true },
```

My project is using jest 27.5.1. If I run `yarn why jest` I get the following:

```
yarn why v1.22.15
[1/4] 🤔  Why do we have the module "jest"...?
[2/4] 🚚  Initialising dependency graph...
[3/4] 🔍  Finding dependency...
[4/4] 🚡  Calculating file sizes...
=> Found "jest@27.5.1"
info Reasons this module exists
   - "_project_#@projectName#backend" depends on it
   - Hoisted from "_project_#@projectName#backend#jest"
   - Hoisted from "_project_#webapp#jest"
info Disk size without dependencies: "48KB"
info Disk size with unique dependencies: "1.11MB"
info Disk size with transitive dependencies: "45.64MB"
info Number of shared dependencies: 172
✨  Done in 0.66s.
```

**Comment by ArtemGovorov** [comment]
@abirtley Thanks for providing the details.

Re: Jest version in the report, it is just a version for our internal use (min version of Jest where auto-config support was introduced for).

Re: being unable to step into the `selectBest` function, can you please share 
- how the function is imported into the test file, 
- where is the file containing the function located relatively to the test file,
- does the file containing the `selectBest` function have green Wallaby gutter indicators?

**Comment by abirtley** [comment]
Sure thing!

## how the function is imported into the test file?

`import { selectBest } from '@projectName/common/tfh';`

## where is the file containing the function located relatively to the test file?

```
- projectName/common/tfh.ts
- projectName/backend/test/common/tfh.test.ts
```

Wallaby config file is located at `projectName/backend`:

```js
module.exports = function () {
  return {
    autoDetect: true,
    testFramework: {
      configFile: './jest.config.js',
    },
  };
};
```

In the same folder is `tsconfig.json` file, containing:

```json
{
  "extends": "../tsconfig.json",
  "include": ["core", "services", "test", "scripts", "../common", "functions"],
  "compilerOptions": {
    "strict": true,
    "baseUrl": ".",
    "paths": {
      "@projectName/backend/core": ["core"],
      "@projectName/backend/core/*": ["core/*"],
      "@projectName/backend/services/common": ["services/common"],
      "@projectName/backend/services/common/*": ["services/common/*"],
      "@projectName/common/*": [
        "../common/*"
      ],
    },
    "target": "es2021",
    "module": "es6",
    "moduleResolution": "node",
    "noEmitHelpers": true,
    "importHelpers": true
  }
}
```

Also in the same folder is `jest.config.js`, which contains the following:

```js
const standardTransform = [
  'esbuild-jest',
  {
    sourcemap: true,
  },
];

module.exports = {
  setupFilesAfterEnv: ['<rootDir>/test/jest.setup.ts'],
  roots: ['<rootDir>/test'],
  testMatch: ['**/*.*test.(ts|tsx|js)'],
  transform: {
    '\\.ts$': standardTransform,
    'true-myth[\\/].*\\.js$': standardTransform,
  },
  moduleNameMapper: {
    '@projectName/common/(.*)': '<rootDir>/../common/$1',
    // we need to manually map sub-module imports for now: https://github.com/facebook/jest/issues/12270
    'true-myth/result': '<rootDir>/../node_modules/true-myth/dist/result.js',
    'true-myth/maybe': '<rootDir>/../node_modules/true-myth/dist/maybe.js',
  },
  // we need to manually transform libraries that apparently export things incorrectly for Jest's liking
  transformIgnorePatterns: ['/node_modules/(?!(true-myth)/)'],
  collectCoverage: true,
  // Note: seems to be a bug in Jest - despite this setting, code coverage
  // is only reported for files under test or referenced in the test suite
  collectCoverageFrom: ['(functions|core|services)/**/*.{ts,js}'],
  // Note: if you have more than 1 worker, then Jest spins up worker child processes,
  // but it *doesn't* pass command line arguments to those subprocesses. That breaks
  // sls-test-tools, which requires things to be passed on the command line.
  // see https://stackoverflow.com/questions/67456849/cant-get-full-process-argv-list-when-passing-custom-arguments-to-jest
  maxWorkers: 1,
};
```

## Does the file containing the selectBest function have green Wallaby gutter indicators?

No it does not! 🕵️ !

Digging into this further, I have some other tests in `/backend/tests` that test code in `/backend/core/`. These do show gutter indicators in the functions under test (ie, in `/backend/core`). And I can 'step into' those functions, as I would expect. So it looks like the file locations are central to the issue.

## Did you read the manual? (my question, not yours)

😨 whoops! I can actually see variable values by highlighting the variable, as [clearly set out in the docs](https://wallabyjs.com/docs/intro/time-travel-debugger.html#inspecting-runtime-values) Not sure why I thought hovering would work. And variable values work even when "step into" does not.

**Comment by ArtemGovorov** [comment]
Thanks for providing the details.

Wallaby debugger can only be used on files instrumented by Wallaby and when deciding what files to instrument the tool respects Jest's `collectCoverageFrom` setting. In your case the setting only lists `functions|core|services` files, and the function that you are trying to step into is defined in a file under `common` dir, so Wallaby doesn't instrument/step into it.

If you extend the `(functions|core|services)/**/*.{ts,js}` pattern to include `common` folder: `functions|core|services|common)/**/*.{ts,js}`, and restart Wallaby, then it should work for you.

**Comment by abirtley** [comment]
We are getting close! I have updated my `jest.config.js` similar to what you suggested, though I had to change my  `rootDir` to be one level up from the `backend` folder. Now if I run the test using command line `jest` I see coverage for all files, including those in the `common` folder. (I was not seeing this before.)

However, Wallaby still does not let me step into functions in the `common` folder. I wonder if Wallaby is not respecting the `rootDir` field? I've copied my config below, and commented the line that I had to add to make anything work in Wallaby. Without this additional line, I couldn't step into any functions in `backend/services`, `backend/core` or `backend/functions`. Note that the jest config file is still stored in `backend`, which is a sibling directory to `common`.

```js
// backend/jest.config.js

module.exports = {
  setupFilesAfterEnv: ['<rootDir>/backend/test/jest.setup.ts'],
  rootDir: path.join(__dirname, '..'),
  testMatch: ['<rootDir>/backend/test/**/*.*test.(ts|tsx|js)'],
  transform: {
    '\\.ts$': standardTransform,
    'true-myth[\\/].*\\.js$': standardTransform,
  },
  moduleNameMapper: {
    '@taleblank/common/(.*)': '<rootDir>/common/$1',
    // we need to manually map sub-module imports for now: https://github.com/facebook/jest/issues/12270
    'true-myth/result': '<rootDir>/node_modules/true-myth/dist/result.js',
    'true-myth/maybe': '<rootDir>/node_modules/true-myth/dist/maybe.js',
  },
  // we need to manually transform libraries that apparently export things incorrectly for Jest's liking
  transformIgnorePatterns: ['/node_modules/(?!(true-myth)/)'],
  collectCoverage: true,
  collectCoverageFrom: [
    'backend/(core|services|functions)/**/*.{ts,js}',
    // the following line is required for wallaby.js to step into files in these folders
    // if you just have the line above, which starts with backend/, wallaby does not step into
    '**/(core|services|functions)/**/*.{ts,js}',
    'common/**/*.{ts,js}',
    // i tried adding the following lines to see if it would make wallaby step into code in the common folder, but they don't help
    '../common/**/*.{ts,js}',
    '**/../common/**/*.{ts,js}',
  ],
  // Note: if you have more than 1 worker, then Jest spins up worker child processes,
  // but it *doesn't* pass command line arguments to those subprocesses. That breaks
  // sls-test-tools, which requires things to be passed on the command line.
  // see https://stackoverflow.com/questions/67456849/cant-get-full-process-argv-list-when-passing-custom-arguments-to-jest
  maxWorkers: 1,
};
```

**Comment by abirtley** [comment]
We are getting close! I have updated my `jest.config.js` similar to what you suggested, though I had to change my  `rootDir` to be one level up from the `backend` folder. Now if I run the test using command line `jest` I see coverage for all files, including those in the `common` folder. (I was not seeing this before.)

However, Wallaby still does not let me step into functions in the `common` folder. I wonder if Wallaby is not respecting the `rootDir` field? I've copied my config below, and commented the line that I had to add to make anything work in Wallaby. Without this additional line, I couldn't step into any functions in `backend/services`, `backend/core` or `backend/functions`. Note that the jest config file is still stored in `backend`, which is a sibling directory to `common`.

```js
// backend/jest.config.js

module.exports = {
  setupFilesAfterEnv: ['<rootDir>/backend/test/jest.setup.ts'],
  rootDir: path.join(__dirname, '..'),
  testMatch: ['<rootDir>/backend/test/**/*.*test.(ts|tsx|js)'],
  transform: {
    '\\.ts$': standardTransform,
    'true-myth[\\/].*\\.js$': standardTransform,
  },
  moduleNameMapper: {
    '@projectName/common/(.*)': '<rootDir>/common/$1',
    // we need to manually map sub-module imports for now: https://github.com/facebook/jest/issues/12270
    'true-myth/result': '<rootDir>/node_modules/true-myth/dist/result.js',
    'true-myth/maybe': '<rootDir>/node_modules/true-myth/dist/maybe.js',
  },
  // we need to manually transform libraries that apparently export things incorrectly for Jest's liking
  transformIgnorePatterns: ['/node_modules/(?!(true-myth)/)'],
  collectCoverage: true,
  collectCoverageFrom: [
    'backend/(core|services|functions)/**/*.{ts,js}',
    // the following line is required for wallaby.js to step into files in these folders
    // if you just have the line above, which starts with backend/, wallaby does not step into
    '**/(core|services|functions)/**/*.{ts,js}',
    'common/**/*.{ts,js}',
    // i tried adding the following lines to see if it would make wallaby step into code in the common folder, but they don't help
    '../common/**/*.{ts,js}',
    '**/../common/**/*.{ts,js}',
  ],
  // Note: if you have more than 1 worker, then Jest spins up worker child processes,
  // but it *doesn't* pass command line arguments to those subprocesses. That breaks
  // sls-test-tools, which requires things to be passed on the command line.
  // see https://stackoverflow.com/questions/67456849/cant-get-full-process-argv-list-when-passing-custom-arguments-to-jest
  maxWorkers: 1,
};
```

**Comment by ArtemGovorov** [comment]
Can you please try this value for the setting:
```
collectCoverageFrom: [
    '**/(core|services|functions|common)/**/*.{ts,js}',
],
```
It should hopefully make both Wallaby and Jest work (also with this value you shouldn't need to change your `rootDir`).

**Comment by abirtley** [comment]
Unfortunately that doesn't work, either on the command line or in wallaby. With that configuration, I think `rootDir` is still `backend`, and the folder structure looks like this (backend and common are sibling folders):
 
```
+ project root
  + common
    - tfh.ts
  + backend
    - jest.config.js
    + test
    + core
    + services
    + functions
    + other folders
```

**Comment by ArtemGovorov** [comment]
Ah, ok, thanks for providing the file tree, I have missed the fact that `backend` and `common` are sibling directories, even though you've been mentioning it. 

Wallaby can only instrument (and step into) files within the working directory that the test run is started. In your case, you are starting Wallaby not inside the `project root` directory, but inside the `backend` directory, so only files in the directory will be instrumented/considered for debugging purposes (if they also satisfy `collectCoverageFrom` pattern).

If you would like Wallaby to instrument files in the `common` directory, you will need to launch Wallaby in your `project root` directory. In order to do it, you'll need to make sure that `npx jest` command is working when running it from your `project root` directory. In order to make it work you may need to create [Jest config with the `projects` setting](https://jestjs.io/docs/configuration#projects-arraystring--projectconfig) in your project root. When Jest CLI starts working from your `project root`, Wallaby should start working too, and you should be able to debug any files under your `project root`.

**Comment by smcenlly** [comment]
I believe @ArtemGovorov's latest comment resolves the problem. Closing this issue. If you're still having problems, please let us know.

**Comment by abirtley** [comment]
Thanks Simon, yes it does - was just about to close it myself.

For anyone else reading this:

* I rejigged the jest configuration so that the backend and webapp jest configurations are now just projects inside a root level jest.config.js. 
* I created a simple wallaby.js file in the root folder, and removed the wallaby.js files from the backend and webapp folders
* I also had to use `ts-jest`'s `ts-config` option so that backend project used the backend tsconfig.json, and webapp project used the webapp tsconfig.json. [Documentation here.](https://kulshekhar.github.io/ts-jest/docs/getting-started/options/tsconfig/) Without that, ts-jest will use a tsconfig.json in your root folder for all projects, which is not what I wanted.
